### PR TITLE
Device Update for IoT Hub GA release updates

### PR DESCRIPTION
**Pull request by dpokluda**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/deviceupdate/azure-iot-deviceupdate/swagger/README.md
Comment (weidongxu-microsoft) [inline sdk/deviceupdate/azure-iot-deviceupdate/swagger/README.md:23]: I think you need to switch to Azure main branch (maybe another regen), before PR can be merged.

Comment (weidongxu-microsoft) [reply sdk/deviceupdate/azure-iot-deviceupdate/swagger/README.md:23]: If you have swagger PR pending, let us know the link and state of it (whether it is approved, or there is change request on it that likely would result in SDK change).

Comment (dpokluda) [reply sdk/deviceupdate/azure-iot-deviceupdate/swagger/README.md]: Updated.

**Comment by weidongxu-microsoft** [comment]
@haolingdong-msft Could you go one round of review, then add Srikanta if it looks good to you?

Since there is no change in ApiView, I think it should be quick for Srikanta.

**Comment by weidongxu-microsoft** [comment]
I've merged the CI change https://github.com/Azure/azure-sdk-for-java/pull/30236

Next time you release azure-iot-deviceupdate, you will see it automatically checked in right panel (and azure-resourcemanager-deviceupdate by default unchecked). This should by default be the correct configure for releasing data-plane SDK.

File: sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md
Comment (haolingdong-msft) [inline sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md:10]: Add dependency updates similar to previous releases:

#### Dependency Updates
- Upgraded `azure-core` from `1.30.0` to version `1.31.0`.
- Upgraded `azure-core-http-netty` from `1.12.3` to version `1.12.4`.

Comment (srnagar) [reply sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md:10]: This is a breaking change. So, we should add this to the "Breaking Changes" section.

Comment (haolingdong-msft) [reply sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md:10]: Also it is suggested to change the content to `Removed optional "filter" query parameter from listBestUpdatesForGroup method in DeviceManagementClient and DeviceManagementAsyncClient`.

Comment (dpokluda) [reply sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md]: Fixed

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

Let us know your release plan, as Sept. monthly release is starting.

**Review by haolingdong-msft — APPROVED** [review]

File: eng/versioning/version_client.txt
Comment (weidongxu-microsoft) [inline eng/versioning/version_client.txt:104]: It should be `com.azure:azure-iot-deviceupdate;1.0.0-beta.3;1.0.0`

First version is one of the last released, second version is the one that to be released next.

Comment (dpokluda) [reply eng/versioning/version_client.txt]: Thank you.

**Review by srnagar — APPROVED** [review]

File: sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md
Comment (srnagar) [inline sdk/deviceupdate/azure-iot-deviceupdate/CHANGELOG.md:3]: nit: update the release date. The release pipeline will block the release if the date is invalid.
```suggestion
## 1.0.0 (2022-09-02)
```

**Review by srnagar — APPROVED** [review]

**Pull request merged**
